### PR TITLE
forgotPasswod-logicAdded

### LINKED ISSUE #65: Forget Password Not working
# Issue Report

## Title

Forget Password Not working

## Description

I’ve noticed that the "Forgot Password" feature isn’t functioning properly. This is critical, as users who forget their passwords risk losing access to their accounts.

It's essential to resolve this issue to ensure a seamless user experience.

## Steps to Reproduce the Issue

1. I will write backend route for forgot password.
2. And further integrate wiith backend.

Thank you for taking the time to report this issue! We appreciate your help in improving our project.

### PR DESCRIPTION
**Pull request by krishpathak**
# Pull Request Template

**Title:** 

Forgot Password logic is added. So, it enables users to change his/her password if he/she forgots

## Description

Frontend Page

-Created a dedicated Forgot Password page that prompts users to enter their email address and new password.
-Implemented input validation to ensure that the new password and confirmation password match before submission.
-The page displays loading indicators while the request is being processed, enhancing user experience.

Backend Route

-Created an Express route that handles incoming password reset requests.
-The route verifies the user's email and updates the password after hashing it for security.
-Implemented error handling to manage various scenarios, such as user not found or internal server errors.

## Related Issue

Fixes #65

## Changes Made

I have successfully developed a "Forgot Password" feature that allows users to reset their passwords securely. This feature includes a user-friendly frontend page and a robust backend route for handling password reset requests.

## Checklist

- [x] I have tested my code and it works as expected.
- [x] I have added/updated documentation where necessary.
- [ ] I have included tests for new features.
- [x] I have followed the project's coding standards.

## Screenshots (if applicable)

[Watch the demonstration](https://github.com/user-attachments/assets/8d3bbf6d-17fa-490c-b714-a5254fa31699)

### CODE REVIEW

**Comment by krishpathak** [comment]
@thakuratul2 @saurabhbakolia

I created a new database entry with the email 'hello@gmail.com', which I'm using in the provided video.

Please take a moment to check the video. The functionality is working perfectly as expected. If you encounter any issues or have any feedback, please let me know!

Thank you!

File: backend/api/controllers/userController.js [hunk 1]
```diff
@@ -90,6 +90,34 @@ exports.sign_in = async function (req, res) {
 			});
 		});
 };
+exports.forgot =async function (req, res, next) {
```
Comment (saurabhbakolia) [inline backend/api/controllers/userController.js:93]: change the function name to forgotPassword.

Comment (krishpathak) [reply backend/api/controllers/userController.js:93]: Please reject this PR I will send it again.

Comment (krishpathak) [reply backend/api/controllers/userController.js]: done

Comment (krishpathak) [reply backend/api/controllers/userController.js]: Please cheak it and if you find any issue revert back to me
Thank You!

**Comment by saurabhbakolia** [comment]
Also don't create a PR to directly push into the `master` branch instead create one for the `develop` branch.

**Comment by krishpathak** [comment]
> Also don't create a PR to directly push into the `master` branch instead create one for the `develop` branch.

I had again sent PR. Reject this PR and you can merge that PR.
Thank You

**Comment by saurabhbakolia** [comment]
Closing the PR, as #74 is a new PR for the same issue.

**Pull request closed without merging**